### PR TITLE
Improve tile documentation

### PR DESCRIPTION
**Pull request by mark-poscablo**
The original documentation didn't really make it clear to me what tile() does, even though it's a fairly simple concept. Here's how I think we could make it clearer.

image

### CODE REVIEW

File: docs/details/data.dox [hunk 1]
```diff
@@ -159,9 +159,31 @@ Requires that all dimensions except the join dimension must be the same for all
 
 \defgroup manip_func_tile tile
 
-\brief Tile the input array along specified dimensions
+\brief Repeat the contents of the input array along the specified dimensions
 
-Creates copys of the array a specified number of times within the output array
+Creates copies of the input array and places them side by side each other, such
+that the output array will have as many copies of the input array as the user
+specifies, along each dimension. In this way, the output array essentially
```
Comment (9prady9) [inline docs/details/data.dox]: I think you can shorten this `In this way, the output array essentially
becomes "tiled", wherein each copy of the input array is a "tile" (including the
input array itself).` further. 

Something like `Essentially, the output array will be a set of tiles where each tile is a copy of input array`.

File: docs/details/data.dox [hunk 2]
```diff
@@ -159,9 +159,31 @@ Requires that all dimensions except the join dimension must be the same for all
...
+
+Here, the input is tiled along the second dimension, 3 times:
+
+\snippet test/tile.cpp ex_tile_1_3
+
+Lastly, one can also tile along multiple dimensions simultaneously. Here, the
+input is tiled 2 times in the first dimension and 3 times in the second
+dimension:
+
+\snippet test/tile.cpp ex_tile_0_2_and_1_3
```
Comment (9prady9) [inline docs/details/data.dox]: :+1:

File: include/af/data.h [hunk 3]
```diff
@@ -234,11 +234,11 @@ namespace af
...
-        \param[in] x is the number of times \p in is tiled along first dimension
-        \param[in] y is the number of times \p in is tiled along second dimension
-        \param[in] z is the number of times \p in is tiled along third dimension
-        \param[in] w is the number of times \p in is tiled along fourth dimension
-        \return the tiled output
+        \param[in] x is how many copies of \p in are tiled along the first dimension
+        \param[in] y is how many copies of \p in are tiled along the second dimension
+        \param[in] z is how many copies of \p in are tiled along the third dimension
+        \param[in] w is how many copies of \p in are tiled along the fourth dimension
+        \return The tiled version of the input array
```
Comment (9prady9) [inline include/af/data.h]: I think mixture of the old and new statement makes more sense, like
`[ x | y | z ] is the number of times \p in is copied along the [ first | second | third ] dimension` what do you think ?

I think this sits better with your earlier explanation in data.dox file.

File: include/af/data.h [hunk 4]
```diff
@@ -247,8 +247,8 @@ namespace af
 
     /**
         \param[in] in is the input array
-        \param[in] dims dim4 of tile dimensions
-        \return the tiled output
+        \param[in] dims specifies how many times \p in is tiled along the dimensions
```
Comment (9prady9) [inline include/af/data.h]: similarly, `\p in is copied along the` instead of `tiled`.

File: include/af/data.h [hunk 5]
```diff
@@ -550,12 +550,12 @@ extern "C" {
...
+        \param[out] out is the tiled version of the input array
         \param[in] in is the input matrix
-        \param[in] x is the number of times \p in is tiled along first dimension
-        \param[in] y is the number of times \p in is tiled along second dimension
-        \param[in] z is the number of times \p in is tiled along third dimension
-        \param[in] w is the number of times \p in is tiled along fourth dimension
+        \param[in] x is how many copies of \p in are tiled along the first dimension
+        \param[in] y is how many copies of \p in are tiled along the second dimension
+        \param[in] z is how many copies of \p in are tiled along the third dimension
+        \param[in] w is how many copies of \p in are tiled along the fourth dimension
```
Comment (9prady9) [inline include/af/data.h]: Similar comment as C++ API documentation about sentence rephrase.

File: docs/details/data.dox [hunk 6]
```diff
@@ -159,9 +159,31 @@ Requires that all dimensions except the join dimension must be the same for all
 
 \defgroup manip_func_tile tile
 
-\brief Tile the input array along specified dimensions
+\brief Repeat the contents of the input array along the specified dimensions
 
-Creates copys of the array a specified number of times within the output array
+Creates copies of the input array and places them side by side each other, such
```
[the remark this answers is not in the document]
Comment (mark-poscablo) [reply docs/details/data.dox]: True, it's the devs that will be reading this most likely anyway.

File: docs/details/data.dox (hunk 1 above)
Comment (mark-poscablo) [reply docs/details/data.dox]: Cool that makes sense, I just mixed your statements with mine.

File: include/af/data.h (hunk 3 above)
Comment (mark-poscablo) [reply include/af/data.h]: So I'm aiming to point out that x, y, z, w count actually includes the original. If we say "the number of times it is copied", my first impression would be how many copies (excluding the original) will be produced.

Comment (mark-poscablo) [reply include/af/data.h]: I do like that wording though, maybe I'll just include a note on the bottom that points out the inclusivity of the counts

**Comment by mark-poscablo** [comment]
Thanks for the feedback @9prady9 ! Here's what it looks like now:

image

Also showing the bottom portion here:

image

**Comment by mark-poscablo** [comment]
Thanks for approving @9prady9, I'll merge this in now

**Pull request merged**